core/lib/zksync_core/src/state_keeper/io/mod.rs

```diff
@@ -99,6 +99,19 @@ impl L1BatchParams {
     }
 }
 
+impl L1BatchParams<()> {
+    pub(crate) fn with_previous_batch_hash(self, previous_batch_hash: H256) -> L1BatchParams {
```

Comment:
...to the point above: especially this transition may be perceived as weird (e.g. the fact that the type of `self` is not `L1BatchParams` (which is not the same as `L1BatchParams<()>`)).